src/magic.cpp

```diff
@@ -195,14 +199,24 @@ const std::string spell_type::sound_variant_default = "default";
 // empty string
 const std::string spell_type::effect_str_default;
 const std::optional<field_type_id> spell_type::field_default = std::nullopt;
-const int spell_type::field_chance_default = 1;
+const float spell_type::field_chance_default = 1.0f;
```

Comment:
Because if field type was 1, it would splatter a field in every single tile. If it was 2, it would splatter a field in 50% of tiles. Making it a float and setting it to 1.9 allows for a small amount of scatter which leads to much better looking field distribution from the boomer spells I've implemented here. This also allows for better field-spreading functionality going forward. The function static casts it, so any spell jsons which are still using integers are unaffected.

Reply:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/94415528/12c4d46a-bedb-40c0-83b2-17cb3a255a19)
Here's an example.  Previously, killing a big boomer would leave a perfect circle of bile every time. Now it looks more like the stuff exploded and splattered around randomly, and since the bile is slippery it also means that it can sometimes leave dry spaces for characters and monsters to navigate without risking a fall.